File: example/src/main/java/com/example/errorview/MainActivity.java
package com.example.errorview;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import butterknife.BindViews;

public class MainActivity extends BaseActivity implements View.OnClickListener {

    @BindViews({R.id.button_main_simple}) Button[] buttons;

    @Override
    public int getContentView() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        for (Button button : buttons)
            button.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button_main_simple:
                startActivity(new Intent(this, SimpleActivity.class));
                break;
        }
    }
}

File: example/src/main/java/com/example/errorview/MainActivity.kt
package com.example.errorview

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.design.widget.BaseTransientBottomBar
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        buttonExample1.setOnClickListener { start(Example1Activity::class.java) }
        buttonExample2.setOnClickListener { start(Example2Activity::class.java) }
    }

    private fun start(cls: Class<out Activity>) {
        startActivity(Intent(this, cls))
    }
}